spark/src/test/scala/org/apache/spark/sql/delta/DeltaTableFeatureSuite.scala

```diff
@@ -205,30 +207,13 @@ class DeltaTableFeatureSuite
     val tfProtocol2 =
       Protocol(TABLE_FEATURES_MIN_READER_VERSION, TABLE_FEATURES_MIN_WRITER_VERSION)
 
-    assert(
-      tfProtocol1.merge(Protocol(1, 2)) ===
-        tfProtocol1.withFeatures(Seq(AppendOnlyTableFeature, InvariantsTableFeature)))
-    assert(
-      tfProtocol2.merge(Protocol(2, 6)) ===
-        tfProtocol2.withFeatures(Set(
-          AppendOnlyTableFeature,
-          InvariantsTableFeature,
-          ColumnMappingTableFeature,
-          ChangeDataFeedTableFeature,
-          CheckConstraintsTableFeature,
-          GeneratedColumnsTableFeature,
-          IdentityColumnsTableFeature,
-          TestLegacyWriterFeature,
-          TestLegacyReaderWriterFeature,
-          TestRemovableLegacyWriterFeature,
-          TestRemovableLegacyReaderWriterFeature)))
+    assert(tfProtocol1.merge(Protocol(1, 2)) === Protocol(1, 2))
+    assert(tfProtocol2.merge(Protocol(2, 6)) === Protocol(2, 6))
   }
 
   test("protocol upgrade compatibility") {
     assert(Protocol(1, 1).canUpgradeTo(Protocol(1, 1)))
     assert(Protocol(1, 1).canUpgradeTo(Protocol(2, 1)))
-    assert(!Protocol(1, 2).canUpgradeTo(Protocol(1, 1)))
-    assert(!Protocol(2, 2).canUpgradeTo(Protocol(2, 1)))
     assert(
```

Comment:
I'm a bit concerned that this needed to be removed in this PR. Intuitively, (1, 2) -> (1,1) is not an "upgrade" so as a developer I expect this to return `false`. Have we changed the definition of `canUpgradeTo` to mean `canTransitionTo`?